pkg/framework/pods.go

```diff
@@ -0,0 +1,13 @@
+package framework
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetPods(client runtimeclient.Client, selector map[string]string) (corev1.PodList, error) {
+	pods := corev1.PodList{}
+	err := client.List(context.TODO(), &pods, runtimeclient.MatchingLabels(selector))
+	return pods, err
+}
```

Comment:
Normally you would return a pointer to an object so that it implements the right interfaces (runtime.Object, metav1.Object), makes it easier to use at the call site
```suggestion
func GetPods(client runtimeclient.Client, selector map[string]string) (*corev1.PodList, error) {
	pods := &corev1.PodList{}
	err := client.List(context.TODO(), pods, runtimeclient.MatchingLabels(selector))
	return pods, err
}
```

Reply:
Thanks for reviewing @JoelSpeed .  This is my first interaction with this specific test frame.  I really do appreciate the guidance here!!  I'll make the appropriate edits.